inc/classes/Addons/Cloudflare/CloudflareFacade.php

```diff
@@ -0,0 +1,260 @@
+<?php
+namespace WP_Rocket\Addons\Cloudflare;
+
+use Cloudflare\Api;
+use Cloudflare\Zone\Cache;
+use Cloudflare\Zone\PageRules;
+use Cloudflare\Zone\Settings;
+use Cloudflare\IPs;
+
+/**
+ * CloudflareFacade
+ *
+ * @since 3.5
+ * @author Soponar Cristina
+ */
+class CloudflareFacade {
+
+	/**
+	 * Cloudflare API Class
+	 *
+	 * @since 3.5
+	 * @author Soponar Cristina
+	 *
+	 * @var Cloudflare\Api
+	 */
+	protected $api;
+
+	/**
+	 * Cloudflare Cache Class
+	 *
+	 * @since 3.5
+	 * @author Soponar Cristina
+	 *
+	 * @var Cloudflare\Zone\Cache
+	 */
+	protected $cache;
+
+	/**
+	 * Cloudflare PageRules Class
+	 *
+	 * @since 3.5
+	 * @author Soponar Cristina
+	 *
+	 * @var Cloudflare\Zone\PageRules
+	 */
+	protected $page_rules;
+
+	/**
+	 * Cloudflare Settings Class
+	 *
+	 * @since 3.5
+	 * @author Soponar Cristina
+	 *
+	 * @var Cloudflare\Zone\Settings
+	 */
+	protected $settings;
+
+	/**
+	 * Cloudflare IPs Class
+	 *
+	 * @since 3.5
+	 * @author Soponar Cristina
+	 *
+	 * @var Cloudflare\IPs
+	 */
+	protected $ips;
+
+	/**
+	 * Cloudflare Zone ID
+	 *
+	 * @since 3.5
+	 * @author Soponar Cristina
+	 *
+	 * @var String
+	 */
+	protected $zone_id;
+
+	/**
+	 * Constructor
+	 *
+	 * @since 3.5
+	 * @author Soponar Cristina
+	 *
+	 * @param Cloudflare\Api            $api        - Cloudflare API Instance.
+	 * @param Cloudflare\Zone\Cache     $cache      - Cloudflare Cache Instance.
+	 * @param Cloudflare\Zone\PageRules $page_rules - Cloudflare Page Rules Instance.
+	 * @param Cloudflare\Zone\Settings  $settings   - Cloudflare Settings Instance.
+	 * @param Cloudflare\IPs            $ips        - Cloudflare IPS Instance.
+	 */
+	public function __construct( $api, $cache, $page_rules, $settings, $ips ) {
+		$this->api        = $api;
+		$this->cache      = $cache;
+		$this->page_rules = $page_rules;
+		$this->settings   = $settings;
+		$this->ips        = $ips;
+	}
+
+	/**
+	 * Sets Cloudflare email, API key and User Agent.
+	 *
+	 * @since  3.5
+	 * @author Soponar Cristina
+	 *
+	 * @param string $email      - Cloudflare Email.
+	 * @param string $api_key    - Cloudflare API Key.
+	 * @param string $zone_id    - Cloudflare Zone ID.
+	 * @param string $user_agent - Cloudflare User Agent.
+	 * @return void
+	 */
+	public function set_api_credentials( $email, $api_key, $zone_id, $user_agent ) {
```

Comment:
You could set the `user_agent` value directly in the method, no need to pass it as a parameter.